backend/api/tests/contract-tests/tests/testClientRegistryContract.spec.ts

```diff
@@ -73,61 +23,121 @@ describe("ClientRegistryProvider", () => {
+            clientId: "testClientId",
+            serviceId: "testServiceId",
+            selfServiceClientId: "testSelfServiceClientId",
+            updates: {
+                ...updatesForClient
+            }
+        }
+
+        const expected_response: clientRegistryResponse  = {
+            client_name: "testClientUpdateResponseName",
+            public_key: "testClientPublicKey",
+            public_key_source: "STATIC",
+            redirect_uris: ["http://testClientUrl"],
+            contacts: updatesForClient.contacts,
+            scopes: ["openid", "email"],
+            subject_type: updatesForClient.subject_type,
+            service_type: "MANDATORY",
+            sector_identifier_uri: "http://gov.uk",
+            client_id: "testClientId",
+            post_logout_redirect_uris: [],
+            back_channel_logout_uri: null,
+            token_endpoint_auth_method: "private_key_jwt",
+            response_type: "code",
+            jar_validation_required: false,
+            claims: [],
+            client_type: "web", 
+            id_token_signing_algorithm: "ES256",
+            jwks_uri: null,
+            channel: "WEB",
+            max_age_enabled: false
+        }
+
+        test("it should return an a client registry object when updating a client", async () => {
             provider
-                .uponReceiving("update a Client")
+                .uponReceiving("configuration to update a client")
                 .withRequest({
                     method: "PUT",
-                    path: "/connect/register",
+                    path: `/connect/register/${putEvent.clientId}`,
                     contentType: "application/json",
                     headers: {
-                        "Content-Type": "application/json; charset=utf-8"
+                        "Content-Type": "application/json;"
                     },
-                    body: EXPECTED_BODY
+                    body: updatesForClient
                 })
                 .willRespondWith({
                     status: 200,
-                    body: eachLike(EXPECTED_BODY),
+                    body: eachLike(expected_response),
                     contentType: "application/json",
                     headers: {
                         "Content-Type": "application/json; charset=utf-8"
                     }
                 });
 
             await provider.executeTest(async mockProvider => {
-                const [results] = await Promise.all([putRequest(mockProvider.url, EXPECTED_BODY)]);
-                console.log("RESULTS: " + JSON.stringify(JSON.stringify(results)));
-                expect(results).toContain(EXPECTED_BODY);
+                process.env.AUTH_REGISTRATION_BASE_URL = mockProvider.url;
+                const results = await updateClientInRegistryHandler(putEvent, mockLambdaContext);
+                expect(JSON.parse(results.body)[0]).toMatchObject(expected_response);
```

Comment:
This is because in the handler it spreads `{ "0": { ....}}` Is this in purpose, it should probably spread the object inside "0".

Reply:
Hmm not sure I understand this one?

Reply:
Hmm not sure I understand this one?

Reply:
We are spreading the object in the handler , but due to the object level we are just spreading a single object `{ "0": { ....}}` instead of whats inside that object. We should probably spread what inside or just set the object with a key instead of `0`